frame/dapps-staking/src/tests.rs

```diff
@@ -1205,3 +1209,418 @@ fn new_era_forcing() {
+    print!("contract_developer_part {:?}\n", expected_developer_reward);
+
+    expected_developer_reward
+}
+
+#[test]
+fn claim_with_more_staking_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        let developer1 = 1;
+        let claimer = 2;
+        const ERA_REWARD: Balance = 1000;
+        const STAKE_AMOUNT: Balance = 100;
+        const INITIAL_STAKE: mock::Balance = 100;
+        let contract =
+            SmartContract::Evm(H160::from_str("1000000000000000000000000000000000000007").unwrap());
+        const FROM_ERA: EraIndex = 1;
+        const SKIP_ERA: EraIndex = 3;
+
+        advance_era_and_reward(FROM_ERA, ERA_REWARD, INITIAL_STAKE);
+        let start_era = DappsStaking::current_era().unwrap_or(Zero::zero());
+        register(developer1, contract.clone());
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, INITIAL_STAKE);
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(1, ERA_REWARD, INITIAL_STAKE);
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, INITIAL_STAKE);
+        let claim_era: EraIndex = DappsStaking::current_era().unwrap_or(Zero::zero());
+        claim(claimer, contract, start_era, claim_era.clone());
+
+        cleared_contract_history(contract, FROM_ERA, claim_era);
+    })
+}
+
+//
+// HELPER FUNCTIONS
+//
+
+// helper fn to make  register() one liner for readability
+fn register(developer: u64, contract: SmartContract<AccountId>) {
+    assert_ok!(DappsStaking::register(
+        Origin::signed(developer),
+        contract.clone()
+    ));
+}
+// helper fn to skip "for_era" eras, but reward each era
+fn advance_era_and_reward(
+    for_era: EraIndex,
+    rewards: BalanceOf<TestRuntime>,
+    staked: BalanceOf<TestRuntime>,
+) {
+    // TODO advance era by incrementing block production, needed for block rewards
+    let current: EraIndex = mock::DappsStaking::current_era().unwrap_or(Zero::zero());
+
+    let era_reward = EraReward { rewards, staked };
+    for era in 0..for_era {
+        <PalletEraRewards<TestRuntime>>::insert(current + era, era_reward.clone());
+    }
+    <CurrentEra<TestRuntime>>::put(&current + for_era);
+}
+
+// helper fn to check updated storage items after claim is called
+fn cleared_contract_history(
+    contract: SmartContract<mock::AccountId>,
+    from_era: EraIndex,
+    to_era: EraIndex,
+) {
+    // check claim pointer moved
+    assert_eq!(
+        mock::DappsStaking::contract_last_claimed(&contract).unwrap_or(Zero::zero()),
+        to_era
+    );
+
+    // check last staked pointer moved
+    assert_eq!(
+        mock::DappsStaking::contract_last_staked(&contract).unwrap_or(Zero::zero()),
+        to_era
+    );
+
+    // check new contractEraStaked
+    assert_ok!((mock::DappsStaking::contract_era_stake(contract, to_era))
+        .is_some()
```

Comment:
You can use standard Rust assert! macro.

assert!(mock::....(contract, to_era).is_some());